### DISCUSSION TITLE
Dependencies folder

### DISCUSSION THREAD

**Discussion by charloulexqs** [BrokenSource/DepthFlow#3] [Q&A] [answered]
Hi, sorry to bother but what is the Dependencies installation folder, cause i have two hard drive i was wondering if it's possible to choose on which one i want them to be saved?

**Comment by Tremeschin** [comment, marked as answer]
Hello, thanks for reaching out!

### Python dependencies
The Python dependencies are managed by [Poetry](https://python-poetry.org/) and by default installed on poetry's [cache directory](https://python-poetry.org/docs/configuration/#cache-directory), I know they get somewhat big but they are configurable with I think `poetry config cache-dir /path/to/cache` or setting the environment variable `POETRY_CACHE_DIR=/path/to/cache (python command)`

### User data
User generated content and caching is currently saved on the "Data" directory based on your OS defined by Python [appdirs](https://github.com/ActiveState/appdirs?tab=readme-ov-file#some-example-output) package, this might change in the future and be more configurable, it's not ideal writing cache (the depth maps to, for example, `~/.local/share/*` directory or `%AppData%/Local/*`), I'm thinking on some local .toml configurable file

### Future of the project
~I'm sorry the project is half baked at this point, I don't fully remember its state also, I'm currently focusing on [ShaderFlow](https://github.com/BrokenSource/ShaderFlow) which has a much better shader render backend but is incomplete and I'm almost happy on the code quality~

~I'm on the final months of graduation and don't have that much free time (more like energy), but stuff will get better soon!~

They did!

**Reply by Tremeschin** [reply]
With code changes since then, it's possible to set the root folder for the Workspace (where the Data, Temp, Downloads, Config, etc. are located, if applicable) is located by setting the `WORKSPACE` environment var:

- `WORKSPACE=/path/to/folder broken depthflow`
